Bottom: 14944660e78363c6bab4cdc351003fe0f8f71bf4
Top:    079d7bbe6129342c62546a42ccf45af677b9160f
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-17 12:46:35 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index ed952d7..6cc0d72 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -836,14 +836,14 @@ pid_t fork(void)
 	ret = real.getsockname(sfd, (struct sockaddr *) &sin6, &len);
 	if (ret)
 		goto out;
-	sin6.sin6_flowinfor = sin6.sin6_scope_id = 0;
+	sin6.sin6_flowinfo = sin6.sin6_scope_id = 0;
 	memset(&sin6.sin6_addr, 0, sizeof sin6.sin6_addr);
 
 	sem = sem_open("/rsocket_fork", O_CREAT, 0644, 1);
 	if (sem == SEM_FAILED)
 		goto out;
 
-	lfd = rsocket(sa.ss_family, SOCK_STREAM, 0);
+	lfd = rsocket(sin6.sin6_family, SOCK_STREAM, 0);
 	if (lfd < 0)
 		goto sclose;
